config/config.go

```diff
@@ -187,30 +185,47 @@ func (c *Config) Validate() error {
 		destinationChains.Add(s.ChainID)
 	}
 
-	return nil
-}
+	// Validate the source chains, and validate that the allowed destinations are configured as destinations
+	sourceChains := set.NewSet[string](len(c.SourceSubnets))
+	for _, s := range c.SourceSubnets {
+		if err := s.Validate(&destinationChains); err != nil {
+			return err
+		}
+		if sourceChains.Contains(s.ChainID) {
+			return fmt.Errorf("configured source subnets must have unique chain IDs")
+		}
+		sourceChains.Add(s.ChainID)
+	}
 
-// GetSourceIDs returns the Subnet and Chain IDs of all subnets configured as a source
-func (cfg *Config) GetSourceIDs() ([]ids.ID, []ids.ID, error) {
+	// Store the source subnet and chain IDs for future use
 	var sourceSubnetIDs []ids.ID
 	var sourceChainIDs []ids.ID
-	for _, s := range cfg.SourceSubnets {
+	for _, s := range c.SourceSubnets {
```

Comment:
Nit: Can we do this in the existing loop over `c.SourceSubnets` above instead of adding a separate iteration?

Reply:
Sure thing. Also clarified some of the surrounding comments.